packages/next-app/src/pages/index.tsx

```diff
@@ -27,23 +39,27 @@ const HomePage: NextPage = () => {
     }
     const variables = toHasuraCondition(query);
     setVariables(variables);
+    setOffset(0);
   }, [query]);
 
   const loadMore = () => {
+    if (isLoading) return;
+    setLoading(true);
     fetchMore({
       variables: {
-        offset: variables.offset + 30,
+        offset: offset + INFINITE_SCROLL_NUMBER,
       },
       updateQuery: (prev, { fetchMoreResult }) => {
         if (!fetchMoreResult) return prev;
+        setOffset(offset + INFINITE_SCROLL_NUMBER);
         return Object.assign({}, prev, {
```

Comment:
do we need object.assign here? 
it seems like we can just use destructuring assignment